File: app/src/main/java/com/lt/ltotc/ui/base/fragment/BasePagerFragment.java
package com.lt.ltotc.ui.base.fragment;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import com.lt.ltotc.BR;
import com.lt.ltotc.R;
import com.lt.ltotc.databinding.FragmentBasePagerBinding;
import com.lt.ltotc.ui.base.adapter.BaseFragmentPagerAdapter;
import com.google.android.material.tabs.TabLayout;

import java.util.List;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import com.lt.basics.base.BaseFragment;
import com.lt.basics.base.BaseViewModel;

/**
 * Created by HankGreen on 2017/7/17.
 * 抽取的二级BasePagerFragment
 */

public abstract class BasePagerFragment extends BaseFragment<FragmentBasePagerBinding, BaseViewModel> {

    private List<Fragment> mFragments;
    private List<String> titlePager;

    protected abstract List<Fragment> pagerFragment();

    protected abstract List<String> pagerTitleString();

    @Override
    public int initContentView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return R.layout.fragment_base_pager;
    }

    @Override
    public int initVariableId() {
        return BR.viewModel;
    }

    @Override
    public void initData() {
        mFragments = pagerFragment();
        titlePager = pagerTitleString();
        //设置Adapter
        BaseFragmentPagerAdapter pagerAdapter = new BaseFragmentPagerAdapter(getChildFragmentManager(), mFragments, titlePager);
        binding.viewPager.setAdapter(pagerAdapter);
        binding.tabs.setupWithViewPager(binding.viewPager);
        binding.viewPager.addOnPageChangeListener(new TabLayout.TabLayoutOnPageChangeListener(binding.tabs));
    }

    @Override
    public void initViewObservable() {

    }
}

File: app/src/main/java/com/lt/ltotc/ui/base/fragment/BasePagerFragment.kt
package com.lt.ltotc.ui.base.fragment

import android.os.Bundle
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.google.android.material.tabs.TabLayout.TabLayoutOnPageChangeListener
import com.lt.basics.base.BaseFragment
import com.lt.basics.base.BaseViewModel
import com.lt.ltotc.BR
import com.lt.ltotc.R
import com.lt.ltotc.databinding.FragmentBasePagerBinding
import com.lt.ltotc.ui.base.adapter.BaseFragmentPagerAdapter

/**
 * Created by HankGreen on 2017/7/17.
 * 抽取的二级BasePagerFragment
 */
abstract class BasePagerFragment : BaseFragment<FragmentBasePagerBinding?, BaseViewModel<*>?>() {
    private var mFragments: List<Fragment>? = null
    private var titlePager: List<String>? = null
    protected abstract fun pagerFragment(): List<Fragment>?
    protected abstract fun pagerTitleString(): List<String>?
    override fun initContentView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): Int {
        return R.layout.fragment_base_pager
    }

    override fun initVariableId(): Int {
        return BR.viewModel
    }

    override fun initData() {
        mFragments = pagerFragment()
        titlePager = pagerTitleString()
        //设置Adapter
        val pagerAdapter = BaseFragmentPagerAdapter(childFragmentManager, mFragments!!, titlePager!!)
        binding!!.viewPager.adapter = pagerAdapter
        binding!!.tabs.setupWithViewPager(binding!!.viewPager)
        binding!!.viewPager.addOnPageChangeListener(TabLayoutOnPageChangeListener(binding!!.tabs))
    }

    override fun initViewObservable() {}
}